tests/api/cross_access/test_model.py

```diff
@@ -0,0 +1,122 @@
+#
+# Copyright 2019 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+"""Test the cross account request model."""
+from api.models import CrossAccountRequest, Employee
+from django.db import IntegrityError, transaction
+from django.test import TestCase
+from django.utils import timezone
+from tenant_schemas.utils import tenant_context
+
+from datetime import timedelta
+from unittest.mock import Mock
+from tests.identity_request import IdentityRequest
+
+
+class CrossAccountRequestModelTests(IdentityRequest):
+    """Test the cross account request model."""
+
+    def setUp(self):
+        """Set up the cross account request model tests."""
+        super().setUp()
+
+        self.ref_time = timezone.now()
+        self.employee = Employee.objects.create(
+            account="654321",
+            username="requestor_test",
+            first_name="requestor",
+            last_name="test",
+            email="requestor@redhat.com",
+        )
+        self.request = CrossAccountRequest.objects.create(
+            target_account="123456", employee=self.employee, endDate=self.ref_time + timedelta(10)
+        )
+
+    def tearDown(self):
+        """Tear down cross account request model tests."""
+        Employee.objects.all().delete()
+        CrossAccountRequest.objects.all().delete()
+
+    def test_request_creation_success(self):
+        """Test the creation of cross account request."""
```

Comment:
When a request is created, is there any validation to confirm that the creator is an internal user, or is that done completely by 3scale?

Also, this is probably my unfamiliarity with Django, but is there any referential integrity we can leverage to make sure that the target account is a real account that exists? If that validation is in-place, then I probably haven't come across it yet ;)

Reply:
That's done by 3scale, it would pass the identity of the creator.

No, me too, I haven't come across any account validation